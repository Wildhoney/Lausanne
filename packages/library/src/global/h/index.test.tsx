import { h } from "./index.js";
import { use } from "../use/index.js";

describe("h()", () => {
  it("should be able to parse simple text children", () => {
    expect(<section>Hi Adam</section>).toMatchSnapshot();
  });

  it("should be able to parse text children with interspersed nodes", () => {
    expect(
      <section>
        Hi <strong>Adam</strong>!
      </section>
    ).toMatchSnapshot();
  });

  it("should be able to parse children with state", () => {
    const [name] = use.state("Adam");

    expect(
      <section>
        Hi <strong>{name()}</strong>!
      </section>
    ).toMatchSnapshot();
  });
});
