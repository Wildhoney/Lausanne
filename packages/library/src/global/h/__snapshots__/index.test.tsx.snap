// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h() should be able to parse children with state 1`] = `
{
  "attrs": null,
  "children": [
    {
      "items": [
        "Hi ",
      ],
      "kind": Symbol(swiss/text),
    },
    {
      "attrs": null,
      "children": [
        {
          "kind": Symbol(swiss/state),
          "value": [Function],
        },
      ],
      "name": "strong",
    },
    {
      "items": [
        "!",
      ],
      "kind": Symbol(swiss/text),
    },
  ],
  "name": "section",
}
`;
